src/foam/lib/formatter/JSONFObjectFormatter.java

```diff
@@ -186,6 +186,32 @@ protected void outputProperty(FObject o, PropertyInfo p) {
     append(':');
     p.formatJSON(this, o);
   }
+
+  protected boolean maybeOutPutFObjectProperty(FObject newFObject, FObject oldFObject, PropertyInfo prop) {
+
+    if ( prop instanceof AbstractFObjectPropertyInfo && oldFObject != null &&
+      prop.get(oldFObject) != null && prop.get(newFObject) != null
+    ) {
+      String before = builder().toString();
```

Comment:
Why do you have to extract 'before'?

Reply:
i have to reset before calling "maybeOutputDelta on line 197, otherwise it would append directly without adding the key

Reply:
You should add a comment saying that for the benefit of future maintainers.